### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by muhammedb8826**
## In this Milestone we have worked on the followings

N + 1 Improvements

User#index before

user index before

User#index after

user index after

User#show before

user show before

User#show after

user show after

- User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.

- user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.

- Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/integration/user_index_spec.rb
Comment (franclobo) [inline spec/integration/user_index_spec.rb:1]: - [ ] Kindly complete all the integration specs for each element of the view, it is a project requirement 👇🏽 

> I can see the username of all other users. ✔️ 
> I can see the profile picture for each user. ✔️ 
> I can see the number of posts each user has written. ✔️ 
> When I click on a user, I am redirected to that user's show page. ❌

File: spec/integration/post_index_spec.rb
Comment (franclobo) [inline spec/integration/post_index_spec.rb:1]: - [ ] Kindly complete all the integration specs required for this project, it lets test all elements of the page 👇🏽 

> I can see the user's profile picture. ❌ 
> I can see the user's username. ✔️ 
> I can see the number of posts the user has written. ✔️ 
> I can see a post's title. ✔️ 
> I can see some of the post's body. ✔️ 
> I can see the first comments on a post.
> I can see how many comments a post has. ✔️ 
> I can see how many likes a post has. ✔️ 
> I can see a section for pagination if there are more posts than fit on the view. ✔️ 
> When I click on a post, it redirects me to that post's show page. ✔️

File: spec/integration/post_show_spec.rb
Comment (franclobo) [inline spec/integration/post_show_spec.rb:1]: - [ ] Kindly complete all the integration specs, this is a project requirement 👇🏽 

> I can see the post's title. ✔️ 
> I can see who wrote the post. ✔️ 
> I can see how many comments it has. ✔️ 
> I can see how many likes it has. ✔️ 
> I can see the post body. ✔️ 
> I can see the username of each commenter. ✔️ 
> I can see the comment each commentor left. ❌

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @muhammedb8826 and @getssh, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- Use Capybara ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] Please 🙏🏽 check all your tests passed 👇🏽 

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/post_show_spec.rb
Comment (muhammedb8826) [reply spec/integration/post_show_spec.rb:1]: Thank you for the feedback I will do that

**Comment by muhammedb8826** [comment]
All the tests are passed

all the tests are passed

**Review by AnselemOdims — APPROVED** [review]
### Hi  @muhammedb8826 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**